controllers/scenario/scenario_controller_test.go

```diff
@@ -189,32 +175,38 @@ var _ = Describe("ScenarioController", func() {
 	})
 
 	It("can Reconcile function prepare the adapter and return the result of the reconcile handling operation", func() {
+		// I have no idea what is this magic but if either of resource deletion is removed, those test won't pass, erroring on:
+		// Message: "integrationtestscenarios.appstudio.redhat.com \"scenario-sample\" already exists"
+		Expect(k8sClient.Delete(ctx, hasScenario)).Should(Succeed())
```

Comment:
So it turned out that just before we initiate reconciler, we would need to delete scenarios. In this case on 2 places.

Reply:
But if we delete it before running reconciler, so how can we make sure it is reconciled according to code https://github.com/redhat-appstudio/integration-service/blob/main/controllers/scenario/scenario_controller.go#L67-L69

Reply:
I see, we tend to delete the resource before reconcile because of the content of ctrl.Request. We have 2 options of dealing with this.
1. Deleting the resource as we do on many places
2. Rename the Name of the object that should be processed by reconcile.

Reply:
I say we stick with the deletion before reconcile, wdyt?

Reply:
I really don't understand how this is supposed to fix test. 
Firstly in BeforeEach we create the scenario and then it's just removed before the useful part of test starts?

Reply:
Would setting the `deletionTimestamp` on the scenario be enough to make this work? That seems preferrable to deleting the resource.

On the other hand, maybe this shouldn't need to be handled as part of this PR as I mentioned in [this comment](https://github.com/redhat-appstudio/integration-service/pull/666/files#r1579168540).

Reply:
Thanks to Kruno for finding the issue, we needed to set the tests to be ordered. Finalizers were taking too much of a time between those deletions, causing the errors.